pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -224,7 +225,7 @@ func (c *Cluster) startProvisioningOverPVCs(config *provisionConfig) {
 	c.DesiredStorage.VolumeSources = append(c.DesiredStorage.VolumeSources, c.prepareStorageClassDeviceSets(config)...)
 
 	// Filtering out valid volume sources
-	validVolumeSources, err := GetValidVolumeSources(c.context.Clientset, c.Namespace, c.DesiredStorage.VolumeSources)
+	validVolumeSources, err := GetValidVolumeSources(c.context.Clientset, c.Namespace, c.isUpgrade, c.DesiredStorage.VolumeSources)
```

Comment:
While you're in here, could you remove the warning on line 237 below? Instead of the warning perhaps all we need is an info message such as "no volume sources defined to configure OSDs on PVCs"